src/hash/hash.c

```diff
@@ -2267,28 +2301,66 @@ void fiftyoneDegreesResultsHashFromEvidence(
 			return;
 		}
 
-		// Values provided are processed based on the Evidence prefix order
-		// of precedence. In the case of Hash, query prefixed evidence should
-		// be used in preference to the header prefix. This supports
-		// situations where a User-Agent that is provided by the calling
-		// application can be used in preference to the one associated with the
-		// calling device.
-		for (int i = 0;
-			i < FIFTYONE_DEGREES_ORDER_OF_PRECEDENCE_SIZE &&
-			results->count == 0;
-			i++) {
+		
+		{
+			stateFragment.results = results;
+			stateFragment.deviceIdsFound = 0;
+			state.state = &stateFragment;
+
 			EvidenceIterate(
 				evidence,
-				prefixOrderOfPrecedence[i],
+				FIFTYONE_DEGREES_EVIDENCE_QUERY,
 				&state,
-				setResultFromEvidence);
+				setResultFromDeviceID);
+
+			if (EXCEPTION_FAILED) {
```

Comment:
I see this fragment copied over and over again 4-5 times in this method, is there a way to reduce this duplication?

Reply:
Refactored:
- https://github.com/postindustria-tech/device-detection-cxx/blob/query-51D-deviceId/src/hash/hash.c#L2386-L2392